imports:
    - { resource: twig.yml }

parameters:
    container.cache: false

    exception.codes:
        InvalidArgumentException: 400
        Sympathy\Db\NotFoundException: 404
        App\Exception\UnauthorizedException: 401
        App\Exception\AccessDeniedException: 403
        Symlex\Router\Exception\AccessDeniedException: 403
        Symlex\Router\Exception\NotFoundException: 404
        Symlex\Router\Exception\MethodNotAllowedException: 405
        App\Exception\FormInvalidException: 409
        Exception: 500

    exception.messages:
        400: 'Bad request'
        401: 'Unauthorized'
        402: 'Payment Required'
        403: 'Forbidden'
        404: 'Not Found'
        405: 'Method Not Allowed'
        406: 'Not Acceptable'
        407: 'Proxy Authentication Required'
        408: 'Request Timeout'
        409: 'Conflict'
        410: 'Gone'
        500: 'Looks like something went wrong!'

services:
    app:
        class: Silex\Application
        public: true

    controller.rest.example:
        class: Symlex\Tests\Kernel\App\ExampleRestController
        public: true

    controller.web.index:
        class: Symlex\Tests\Kernel\App\ExampleWebController
        public: true

    router.rest:
        class: Symlex\Router\RestRouter
        public: true
        arguments: [ "@app", "@service_container" ]

    router.twig:
        class: Symlex\Router\TwigRouter
        public: true
        arguments: [ "@app", "@service_container", "@twig" ]

    router.twig_default:
        class: Symlex\Router\TwigDefaultRouter
        public: true
        arguments: [ "@app", "@service_container", "@twig" ]

    router.error:
        class: Symlex\Router\ErrorRouter
        public: true
        arguments: [ "@app", "@twig", "%exception.codes%", "%exception.messages%", "%app.debug%" ]